widgetssdk/src/main/java/com/glia/widgets/chat/controller/ChatController.java

```diff
@@ -1412,6 +1409,9 @@ public void newEngagementLoaded(OmnicoreEngagement engagement) {
             Logger.d(TAG, "unsentMessage sent!");
         }
         emitViewState(chatState.engagementStarted());
+        // Loading chat history again on engagement start in case it was an-authenticated visitor that restored ongoing engagement
+        // Currently there is no direct way to know if Visitor is authenticated.
```

Comment:
We can use `authentication.isAuthenticated()` to know if Visitor is authenticated.